core/src/test/scala/org/broadinstitute/dsde/rawls/user/UserServiceSpec.scala

```diff
@@ -401,4 +404,132 @@ class UserServiceSpec extends AnyFlatSpecLike with TestDriverComponent with Mock
     }
   }
 
+  it should "update the billing account for a billing project" in {
+    withMinimalTestDatabase { dataSource: SlickDataSource =>
+      val billingProject = minimalTestData.billingProject
+      val billingAccountName = RawlsBillingAccountName("new-billing-account")
+      val googleProjectName = GoogleProject(billingProject.projectName.value) //TODO: See CA-1363 for PPW concerns
+      val newBillingAccountRequest = UpdateRawlsBillingAccountRequest(billingAccountName)
+
+      val mockSamDAO = mock[SamDAO](RETURNS_SMART_NULLS)
+      when(mockSamDAO.userHasAction(SamResourceTypeNames.billingProject, billingProject.projectName.value, SamBillingProjectActions.updateBillingAccount, userInfo)).thenReturn(Future.successful(true))
+
+      val mockGcsDAO = mock[GoogleServicesDAO](RETURNS_SMART_NULLS)
+      when(mockGcsDAO.setGoogleProjectBillingAccount(ArgumentMatchers.eq(googleProjectName), any[Option[RawlsBillingAccountName]], any[UserInfo])(any[ExecutionContext])).thenReturn(Future.unit)
+      when(mockGcsDAO.testBillingAccountAccess(ArgumentMatchers.eq(billingAccountName), any[UserInfo])).thenReturn(Future.successful(true))
+
+      val userService = getUserService(dataSource, mockSamDAO, mockGcsDAO)
+
+      Await.result(userService.updateBillingProjectBillingAccount(billingProject.projectName, newBillingAccountRequest), Duration.Inf) shouldEqual ()
+
+      val spendReportDatasetInDb = runAndWait(dataSource.dataAccess.rawlsBillingProjectQuery.filter(_.projectName === billingProject.projectName.value).map(row => (row.spendReportDataset, row.spendReportTable)).result)
+
+      //assert that the spend report configuration has been fully cleared
+      spendReportDatasetInDb.head shouldEqual (None, None)
+
+      verify(mockGcsDAO, times(1)).setGoogleProjectBillingAccount(ArgumentMatchers.eq(googleProjectName), any[Option[RawlsBillingAccountName]], any[UserInfo])(any[ExecutionContext])
+    }
+  }
```

Comment:
Matt, I know you pulled all this over from the PPW branch, but I am inclined to add a test for when we call the Update Billing Account endpoint with an empty/blank Billing Account field.  At the moment, I think that's something somebody could do and I think it would result in a failure when checking if we have access to the Billing Account, and I guess that's fine, but that's the sort of thing we should probably test because callers of the API may come to depend on that behavior.

Reply:
These tests were actually not lifted from the PPW branch. I can look at adding that test case, though.

Reply:
I am writing a test for this scenario and truthfully I don't think that it's valuable. This would test Google's behavior, which I can certainly mock out, but the assertion would just be that the mock did what I told it to do.

Reply:
That makes sense.  I guess from my perspective, the long story short is that I'd rather see us have some kind of validation check(s) on the Billing Account that was submitted.  And whether we do that is based on standards/expectations we already set for other fields maybe as well as how the code behaves today.  

What validations should we check?  Do we know the min/max lengths and maybe regex that a Billing Account should match?  We certainly know that if it's an empty string, it's not valid, and I'd rather explicitly catch that.  

My preference would be that overall, we do good validation on values entered rather than counting on making the call to Google to catch whether we entered a bad value.  But I'm ok leaving it up to you to decide if you want to add this at this time.  It wasn't specified in the original ticket, and we can add it later if people are concerned about the behavior or the UX proves problematic.  In general, I do just want to put in my plug for doing good validation when we can.